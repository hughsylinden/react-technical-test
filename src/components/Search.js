import "../styles/Search.css";
import React, { useState } from "react";

export default function Search() {
  const [search, setSearch] = useState("");

  const handleSearchInput = (e) => {
    setSearch(e.target.value);
  };
  return (
    <>
      <form className="search-form">
        <input
          className="search-input"
          type="text"
          data-testid="search-bar"
          onChange={handleSearchInput}
        />
        <button className="search-button" type="submit" name="search-name">
          search
        </button>
      </form>
      {search}
    </>
  );
}
