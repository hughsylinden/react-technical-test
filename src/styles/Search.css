.search-input{
  height: 50px;
  border-radius: 11px;
  font-size: 20px;
  padding: 0 10px;
}

.search-form{
  margin: auto;
  display:flex;
  flex-direction: row;
}

.search-button{
  border-style: solid;
  border-radius: 11px;
}