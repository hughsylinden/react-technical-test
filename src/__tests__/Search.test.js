import React from "react";
import { render } from "@testing-library/react";
import Search from "../components/Search";

describe('Search', () => {

  it('renders correctly', ()=> {
    const { getByTestId, asFragment } = render(<Search />);
    expect(asFragment()).toMatchSnapshot();
    expect(getByTestId('search-bar')).toHaveClass("search-input");
  })
})
