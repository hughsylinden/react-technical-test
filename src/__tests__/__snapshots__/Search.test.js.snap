// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <form
    class="search-form"
  >
    <input
      class="search-input"
      data-testid="search-bar"
      type="text"
    />
    <button
      class="search-button"
      name="search-name"
      type="submit"
    >
      search
    </button>
  </form>
</DocumentFragment>
`;
